flake.nix

```diff
@@ -19,38 +19,34 @@
       systems = [ "x86_64-linux" "aarch64-linux" "aarch64-darwin" "x86_64-darwin" ];
       perSystem = { config, pkgs, system, inputs', ... }:
         let
-          default_gcc = { cross ? true }:
-            let
-              glibc-join = p: p.buildPackages.symlinkJoin {
-                name = "glibc-join";
-                paths = [ p.glibc p.glibc.static ];
-              };
+          glibc-join = p: p.buildPackages.symlinkJoin {
+            name = "glibc-join";
+            paths = [ p.glibc p.glibc.static ];
+          };
 
-              wrap-gcc = p: p.buildPackages.wrapCCWith {
-                cc = p.buildPackages.gcc13.cc;
-                bintools = p.buildPackages.wrapBintoolsWith {
-                  bintools = p.buildPackages.binutils-unwrapped;
-                  libc = glibc-join p;
-                };
-              };
+          wrap-gcc = p: p.buildPackages.wrapCCWith {
+            cc = p.buildPackages.gcc13.cc;
+            bintools = p.buildPackages.wrapBintoolsWith {
+              bintools = p.buildPackages.binutils-unwrapped;
+              libc = glibc-join p;
+            };
+          };
 
+          native-gcc =
+            if pkgs.stdenv.isDarwin
+            then null
+            else wrap-gcc pkgs;
+
+          # cross is for determining whether to install the cross toolchain or not
+          core = { cross ? true }:
+            let
               x86_64-gcc = wrap-gcc pkgs.pkgsCross.gnu64;
               aarch64-gcc = wrap-gcc pkgs.pkgsCross.aarch64-multiplatform;
             in
-            if pkgs.stdenv.isDarwin
-            then [ ]
-            else if cross
-            then if pkgs.stdenv.isAarch64
+            if (cross && !pkgs.stdenv.isDarwin)
```

Comment:
This is because we only support cross compilation on non-Mac AArch64, right? We don't support Mac cross compilation for x86, nor AArch64 cross compilation on x86.

If that's true, can you add a comment to this effect?

Can you also comment on why in the cross case we list two compilers, and whether the order matters?

Reply:
That's understood and fine. Let's add a comment and merge.

Reply:
In the cross case, although listing the one that is not the native toolchain should be sufficient, but that would require adding some platform checks. I personally think that's ugly :)

And nix should evaluate correctly that one of the toolchains in the list is exactly the same as the one `native-gcc` specified, thus not introducing redundant native toolchain.

Reply:
@hanno-becker done